### PR TITLE
[Vulnerability Patch] CWE-471 caused by handlebars@4.0.12

### PR DESCRIPTION
**Pull request by MichaelCherry**
# Description

This patch solves [Prototype Pollution](https://app.snyk.io/vuln/SNYK-JS-HANDLEBARS-173692), related to [CWE-471](https://cwe.mitre.org/data/definitions/471.html) caused by handlebars@4.0.12. Also fixed other dependencies to remove errors and warnings during compilation.

Changes:
- handlebars 4.0.12 -> 4.1.0 (solves High Vulnerability Prototype Pollution)
- react-scripts 2.1.3 -> 2.1.5 (2.1.3 requires earlier version of handlebars)
- added typescript (missing peer dependency for react-scripts)
- removed all devDependencies as unnecessary (versions of eslint conflicted with other deps)

## Type of change

Please delete options that are not relevant.

- Dependency change

## Screenshot

Image of snyk test for frontend after patch
image

# Testing

Applicable for all code changes.

Manual steps to reproduce this functionality:

1.  Delete frontend/node_modules folder
2. Run docker-compose up
3. Verify that application works as expected
4. Verify that prettier/eslint are still working

Any concerns that need to be addressed in product usability testing:

- I did remove eslint and prettier from devDependencies, but this does not seem to have any effect on the auto-reformatting or any part of our application. Just something to keep in mind, though.

Screenshot of unit tests run passing:
image

# Checklist

Applicable for all code changes.

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [x] My changes generate no new compiler warnings
- [x] My changes generate no new accessibility errors and/or warnings
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [x] New and existing unit tests pass locally with my changes
- [x] I have researched WCAG2.1 accessibility standards and met them in this PR (can be navigated using a keyboard)
- [x] My changes look good on IE 10+, Firefox, and Chrome

### CODE REVIEW

**Comment by MichaelCherry** [comment]
@caleybrock When you do review this, please look at the detail link to verify that the vulnerability is gone when run in the online tool. Thanks!
image

**Comment by fnormand01** [comment]
I did the steps you put in the description. The app was running as expected. I also ran the snyk tests and the front end has not vulnerability, as expected.

I'm fine with these updates.

File: frontend/package.json [hunk 1]
```diff
@@ -32,15 +34,5 @@
     "not dead",
     "not ie <= 8",
     "not op_mini all"
-  ],
-  "devDependencies": {
-    "eslint": "5.6.0",
```
Comment (caleybrock) [inline frontend/package.json]: I don't understand why we are removing these. They provide important errors/warnings that don't show up in our editor, and allow us to not use the same editor and get the same warnings.

Comment (MichaelCherry) [reply frontend/package.json]: Give me a couple minutes and I'll send you the warnings/errors that occur when I leave them in

Comment (MichaelCherry) [reply frontend/package.json]: with all the devdeps in place:
image
image

**Pull request merged**